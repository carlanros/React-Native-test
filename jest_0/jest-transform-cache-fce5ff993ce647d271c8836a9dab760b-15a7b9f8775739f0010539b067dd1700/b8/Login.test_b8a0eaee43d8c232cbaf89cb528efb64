4c194afd82a54addeabc670ae54c44fd
'use strict';require('ts-jest').install("/Users/carlanros/React-Native-test/source/components/Login/Login.test.tsx", "import React from 'react';\nimport { TextInput } from 'react-native';\nimport { shallow } from 'enzyme';\nimport Login from '../Login';\njest.unmock('react-native');\nfunction setup() {\n    const props = {};\n    const enzymeWrapper = shallow(<Login {...props}/>);\n    return {\n        props,\n        enzymeWrapper,\n    };\n}\ndescribe('components', () => {\n    describe('Login', () => {\n        it('should render self and subcomponents', () => {\n            const { enzymeWrapper } = setup();\n            expect(enzymeWrapper).toMatchSnapshot();\n            const textInput2 = enzymeWrapper.find(TextInput).at(1).props();\n            expect(textInput2.placeholder).toBe('Password');\n        });\n    });\n});\n//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiTG9naW4udGVzdC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbIkxvZ2luLnRlc3QudHN4Il0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUNBLE9BQU8sS0FBSyxNQUFNLE9BQU8sQ0FBQztBQUMxQixPQUFPLEVBQUMsU0FBUyxFQUFDLE1BQU0sY0FBYyxDQUFDO0FBQ3ZDLE9BQU8sRUFBRSxPQUFPLEVBQUUsTUFBTSxRQUFRLENBQUM7QUFFakMsT0FBTyxLQUFLLE1BQU0sVUFBVSxDQUFDO0FBRTdCLElBQUksQ0FBQyxNQUFNLENBQUMsY0FBYyxDQUFDLENBQUM7QUFFNUI7SUFDRSxNQUFNLEtBQUssR0FBRyxFQUdiLENBQUM7SUFFRixNQUFNLGFBQWEsR0FBRyxPQUFPLENBQUMsQ0FBQyxLQUFLLENBQUMsSUFBSSxLQUFLLENBQUMsRUFBRyxDQUFFLENBQUM7SUFFckQsTUFBTSxDQUFDO1FBQ0wsS0FBSztRQUNMLGFBQWE7S0FDZCxDQUFDO0FBQ0osQ0FBQztBQUVELFFBQVEsQ0FBQyxZQUFZLEVBQUUsR0FBRyxFQUFFO0lBTzFCLFFBQVEsQ0FBQyxPQUFPLEVBQUUsR0FBRyxFQUFFO1FBQ3JCLEVBQUUsQ0FBQyxzQ0FBc0MsRUFBRSxHQUFHLEVBQUU7WUFDOUMsTUFBTSxFQUFFLGFBQWEsRUFBRSxHQUFHLEtBQUssRUFBRSxDQUFDO1lBQ2xDLE1BQU0sQ0FBQyxhQUFhLENBQUMsQ0FBQyxlQUFlLEVBQUUsQ0FBQztZQUd4QyxNQUFNLFVBQVUsR0FBRyxhQUFhLENBQUMsSUFBSSxDQUFDLFNBQVMsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxDQUFDLENBQUMsQ0FBQyxLQUFLLEVBQUUsQ0FBQztZQUMvRCxNQUFNLENBQUMsVUFBVSxDQUFDLFdBQVcsQ0FBQyxDQUFDLElBQUksQ0FBQyxVQUFVLENBQUMsQ0FBQztRQUNsRCxDQUFDLENBQUMsQ0FBQztJQUNMLENBQUMsQ0FBQyxDQUFDO0FBQ0wsQ0FBQyxDQUFDLENBQUMiLCJzb3VyY2VzQ29udGVudCI6WyIvLy8gPHJlZmVyZW5jZSB0eXBlcz1cImplc3RcIi8+XG5pbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IHtUZXh0SW5wdXR9IGZyb20gJ3JlYWN0LW5hdGl2ZSc7XG5pbXBvcnQgeyBzaGFsbG93IH0gZnJvbSAnZW56eW1lJztcblxuaW1wb3J0IExvZ2luIGZyb20gJy4uL0xvZ2luJztcblxuamVzdC51bm1vY2soJ3JlYWN0LW5hdGl2ZScpO1xuXG5mdW5jdGlvbiBzZXR1cCgpIHtcbiAgY29uc3QgcHJvcHMgPSB7XG4vLyBIw6RyIGthbiBuaSBtb2NrYSBrb21wb25lbnRlbnMgcHJvcHMgb20gc8OlZGFuYSBmaW5uc1xuLy8gdHlwIHVzZXJOYW1lOiAndGVzdCdcbiAgfTtcblxuICBjb25zdCBlbnp5bWVXcmFwcGVyID0gc2hhbGxvdyg8TG9naW4gey4uLnByb3BzfSAvPiApO1xuXG4gIHJldHVybiB7XG4gICAgcHJvcHMsXG4gICAgZW56eW1lV3JhcHBlcixcbiAgfTtcbn1cblxuZGVzY3JpYmUoJ2NvbXBvbmVudHMnLCAoKSA9PiB7XG4gIC8qKlxuICAgKiBUZXN0IGNvbXBvbmVudCByZW5kZXJpbmcuIFByb3BlcnRpZXMgb2YgY2hpbGRyZW4gbWlnaHQgYmUgdGVzdGVkIGJ5IGltcG9ydGluZyB0aGVpciB0eXBlOlxuICAgKiAgIGltcG9ydCB7IFRleHQgfSBmcm9tICdyZWFjdC1uYXRpdmUnO1xuICAgKiAgICBjb25zdCBidXN5UHJvcHMgPSBlbnp5bWVXcmFwcGVyLmZpbmQoVGV4dCkucHJvcHMoKTtcbiAgICogICAgZXhwZWN0KGJ1c3lQcm9wcy50ZXh0KS50b0JlKCd0ZXN0KTtcbiAgICovXG4gIGRlc2NyaWJlKCdMb2dpbicsICgpID0+IHtcbiAgICBpdCgnc2hvdWxkIHJlbmRlciBzZWxmIGFuZCBzdWJjb21wb25lbnRzJywgKCkgPT4ge1xuICAgICAgY29uc3QgeyBlbnp5bWVXcmFwcGVyIH0gPSBzZXR1cCgpO1xuICAgICAgZXhwZWN0KGVuenltZVdyYXBwZXIpLnRvTWF0Y2hTbmFwc2hvdCgpO1xuXG4gICAgICAvLyBUZXN0IExvZ2luXG4gICAgICBjb25zdCB0ZXh0SW5wdXQyID0gZW56eW1lV3JhcHBlci5maW5kKFRleHRJbnB1dCkuYXQoMSkucHJvcHMoKTtcbiAgICAgIGV4cGVjdCh0ZXh0SW5wdXQyLnBsYWNlaG9sZGVyKS50b0JlKCdQYXNzd29yZCcpO1xuICAgIH0pO1xuICB9KTtcbn0pOyJdfQ==");'use strict';



jest.unmock('react-native');var _react=require('react');var _react2=_interopRequireDefault(_react);var _reactNative=require('react-native');var _enzyme=require('enzyme');var _Login=require('../Login');var _Login2=_interopRequireDefault(_Login);function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}
function setup(){
var props={};
var enzymeWrapper=(0,_enzyme.shallow)(_react2.default.createElement(_Login2.default,props));
return{
props:props,
enzymeWrapper:enzymeWrapper};

}
describe('components',function(){
describe('Login',function(){
it('should render self and subcomponents',function(){var _setup=
setup(),enzymeWrapper=_setup.enzymeWrapper;
expect(enzymeWrapper).toMatchSnapshot();
var textInput2=enzymeWrapper.find(_reactNative.TextInput).at(1).props();
expect(textInput2.placeholder).toBe('Password');
});
});
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiTG9naW4udGVzdC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbIkxvZ2luLnRlc3QudHN4Il0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7Ozs7QUFPQSxBQUFJLEtBQUMsQUFBTSxBQUFDLEFBQWMsQUFBQyxBQUFDLHVCQU41QixBQUFPLEFBQUssQUFBTSxBQUFPLEFBQUMsdUVBQzFCLEFBQU8sQUFBQyxBQUFTLEFBQUMsQUFBTSxBQUFjLEFBQUMseUNBQ3ZDLEFBQU8sQUFBRSxBQUFPLEFBQUUsQUFBTSxBQUFRLEFBQUMsOEJBRWpDLEFBQU8sQUFBSyxBQUFNLEFBQVUsQUFBQztBQUk3QjtBQUNFLEdBQU0sQUFBSyxPQUFHLEFBR2IsQUFBQztBQUVGLEdBQU0sQUFBYSxlQUFHLEFBQU8sb0JBQUMsQUFBQyxBQUFLLEFBQUMsOENBQUksQUFBSyxBQUFDLEFBQUcsQUFBRSxBQUFDO0FBRXJELEFBQU0sTUFBQztBQUNMLEFBQUs7QUFDTCxBQUFhLEFBQ2QsQUFBQyxBQUNKOztBQUFDO0FBRUQsQUFBUSxTQUFDLEFBQVksYUFBRSxBQUFHLEFBQUU7QUFPMUIsQUFBUSxTQUFDLEFBQU8sUUFBRSxBQUFHLEFBQUU7QUFDckIsQUFBRSxHQUFDLEFBQXNDLHVDQUFFLEFBQUcsQUFBRTtBQUNwQixBQUFLLEFBQUUsQUFBQyxPQUFsQyxBQUFNLENBQUUsQUFBYSxBQUFFO0FBQ3ZCLEFBQU0sT0FBQyxBQUFhLEFBQUMsZUFBQyxBQUFlLEFBQUUsQUFBQztBQUd4QyxHQUFNLEFBQVUsWUFBRyxBQUFhLGNBQUMsQUFBSSxBQUFDLEFBQVMsQUFBQyw2QkFBQyxBQUFFLEdBQUMsQUFBQyxBQUFDLEdBQUMsQUFBSyxBQUFFLEFBQUM7QUFDL0QsQUFBTSxPQUFDLEFBQVUsV0FBQyxBQUFXLEFBQUMsYUFBQyxBQUFJLEtBQUMsQUFBVSxBQUFDLEFBQUMsQUFDbEQ7QUFBQyxBQUFDLEFBQUMsQUFDTDtBQUFDLEFBQUMsQUFBQyxBQUNMO0FBQUMsQUFBQyxBQUFDIiwic291cmNlc0NvbnRlbnQiOlsiLy8vIDxyZWZlcmVuY2UgdHlwZXM9XCJqZXN0XCIvPlxuaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCB7VGV4dElucHV0fSBmcm9tICdyZWFjdC1uYXRpdmUnO1xuaW1wb3J0IHsgc2hhbGxvdyB9IGZyb20gJ2VuenltZSc7XG5cbmltcG9ydCBMb2dpbiBmcm9tICcuLi9Mb2dpbic7XG5cbmplc3QudW5tb2NrKCdyZWFjdC1uYXRpdmUnKTtcblxuZnVuY3Rpb24gc2V0dXAoKSB7XG4gIGNvbnN0IHByb3BzID0ge1xuLy8gSMOkciBrYW4gbmkgbW9ja2Ega29tcG9uZW50ZW5zIHByb3BzIG9tIHPDpWRhbmEgZmlubnNcbi8vIHR5cCB1c2VyTmFtZTogJ3Rlc3QnXG4gIH07XG5cbiAgY29uc3QgZW56eW1lV3JhcHBlciA9IHNoYWxsb3coPExvZ2luIHsuLi5wcm9wc30gLz4gKTtcblxuICByZXR1cm4ge1xuICAgIHByb3BzLFxuICAgIGVuenltZVdyYXBwZXIsXG4gIH07XG59XG5cbmRlc2NyaWJlKCdjb21wb25lbnRzJywgKCkgPT4ge1xuICAvKipcbiAgICogVGVzdCBjb21wb25lbnQgcmVuZGVyaW5nLiBQcm9wZXJ0aWVzIG9mIGNoaWxkcmVuIG1pZ2h0IGJlIHRlc3RlZCBieSBpbXBvcnRpbmcgdGhlaXIgdHlwZTpcbiAgICogICBpbXBvcnQgeyBUZXh0IH0gZnJvbSAncmVhY3QtbmF0aXZlJztcbiAgICogICAgY29uc3QgYnVzeVByb3BzID0gZW56eW1lV3JhcHBlci5maW5kKFRleHQpLnByb3BzKCk7XG4gICAqICAgIGV4cGVjdChidXN5UHJvcHMudGV4dCkudG9CZSgndGVzdCk7XG4gICAqL1xuICBkZXNjcmliZSgnTG9naW4nLCAoKSA9PiB7XG4gICAgaXQoJ3Nob3VsZCByZW5kZXIgc2VsZiBhbmQgc3ViY29tcG9uZW50cycsICgpID0+IHtcbiAgICAgIGNvbnN0IHsgZW56eW1lV3JhcHBlciB9ID0gc2V0dXAoKTtcbiAgICAgIGV4cGVjdChlbnp5bWVXcmFwcGVyKS50b01hdGNoU25hcHNob3QoKTtcblxuICAgICAgLy8gVGVzdCBMb2dpblxuICAgICAgY29uc3QgdGV4dElucHV0MiA9IGVuenltZVdyYXBwZXIuZmluZChUZXh0SW5wdXQpLmF0KDEpLnByb3BzKCk7XG4gICAgICBleHBlY3QodGV4dElucHV0Mi5wbGFjZWhvbGRlcikudG9CZSgnUGFzc3dvcmQnKTtcbiAgICB9KTtcbiAgfSk7XG59KTsiXX0=