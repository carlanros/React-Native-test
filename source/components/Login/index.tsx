// login/index
import React from  'react';
import { View, Text, TextInput } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { connect, Dispatch} from 'react-redux';
import * as types from '../../Types';
import { FacebookLogin } from './actions';
import { bindActionCreators }from 'redux';
import { SocialIcon } from 'react-native-elements';
import { FacebookLogin } from './actions/index';

export class Login extends React.Component<types.IProps, {}> {
    render() {
        return (
            <View style={{padding: 40}}>
                <Text>Login</Text>
                <SocialIcon
                    title='Sign In With Facebook'
                    button
                    onPress={this.facebookLogin}
                    type='facebook'
                    />
            </View>
       );
    }
facebookLogin = async () => {
    await this.props.FacebookLogin();
    if (this.props.login.isLoggedIn) {
        Actions.pop();
    }
}

onTextChanged = (text: string) => {
        this.userName = text;
        if (text && text.length > 5) { Actions.pop(); }
    }
    userName: string = 'unknown';

}
const mapStateToProps = (state: types.IApplicationState) => ({
   // route: state.route,
   login: state.login,
  });

  const mapDispatchToProps = (dispatch: Dispatch<types.IProps>) => ({
    dispatch,
    FacebookLogin: bindActionCreators(FacebookLogin, dispatch),

  });
export default connect<types.IApplicationState, types.IProps>(
mapStateToProps,
mapDispatchToProps
)(Login);