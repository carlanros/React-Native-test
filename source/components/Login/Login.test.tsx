/// <reference types="jest"/>
import React from 'react';
import {TextInput} from 'react-native';
import { shallow } from 'enzyme';

import {Login} from '../Login';

jest.unmock('react-native');

function setup() {
  const props = {
// Här kan ni mocka komponentens props om sådana finns
// typ userName: 'test'
  };

  const enzymeWrapper = shallow(<Login {...props} /> );

  return {
    props,
    enzymeWrapper,
  };
}

describe('components', () => {
  /**
   * Test component rendering. Properties of children might be tested by importing their type:
   *   import { Text } from 'react-native';
   *    const busyProps = enzymeWrapper.find(Text).props();
   *    expect(busyProps.text).toBe('test);
   */
  describe('Login', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();

      // Test Login
      const textInput2 = enzymeWrapper.find(TextInput).at(0).props();
      expect(textInput2.placeholder).toBe('User');
    });
  });
});