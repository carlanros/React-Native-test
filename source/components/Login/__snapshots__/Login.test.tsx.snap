// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Login should render self and subcomponents 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    login={
      Object {
        "isLoggedIn": false,
        "user": null,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Login
        </Text>,
        <UNDEFINED
          button={true}
          onPress={[Function]}
          title="Sign In With Facebook"
          type="facebook"
        />,
      ],
      "style": Object {
        "padding": 40,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Login",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "Login",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "button": true,
          "onPress": [Function],
          "title": "Sign In With Facebook",
          "type": "facebook",
        },
        "ref": null,
        "rendered": null,
        "type": undefined,
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Login
          </Text>,
          <UNDEFINED
            button={true}
            onPress={[Function]}
            title="Sign In With Facebook"
            type="facebook"
          />,
        ],
        "style": Object {
          "padding": 40,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Login",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "Login",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "button": true,
            "onPress": [Function],
            "title": "Sign In With Facebook",
            "type": "facebook",
          },
          "ref": null,
          "rendered": null,
          "type": undefined,
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
